datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -950,30 +909,12 @@ macro_rules! assert_contains {
     };
 }
 
-/// Apply simplification and constant propagation to ([Expr]).
```

Comment:
Instead of adding a `simplify` method on to `Expr` via this trait, I propose to have an `ExprSimplifier` struct that has a simplify method.


cc @ygf11
I found it made the examples in `expr_api.rs` less awkward to write because the schema wasn't needed

Reply:
Looks great. It is more flexible to use, since user can define their `SimplifyInfo` other than inner `SimplifyContext` now.